src/zksync/compile/project.rs

```diff
@@ -0,0 +1,442 @@
+use crate::{
+    artifact_output::{ArtifactOutput, Artifacts},
+    artifacts::{VersionedFilteredSources, VersionedSources},
+    config::ProjectPathsConfig,
+    error::Result,
+    filter::SparseOutputFilter,
+    resolver::GraphEdges,
+    zksync::{
+        artifact_output::zk::ZkContractArtifact,
+        artifacts::{CompilerInput, Settings},
+        cache::ArtifactsCache,
+        compile::{
+            output::{AggregatedCompilerOutput, ProjectCompileOutput},
+            ZkSolc,
+        },
+    },
+    Graph, Project, Solc, Sources,
+};
+use std::{collections::BTreeMap, path::PathBuf};
+
+#[derive(Debug)]
+// NOTE(We need the root ArtifactOutput because of the Project type
+// but we are not using to compile anything zksync related)
```

Comment:
nit: Not sure if this is copied code. We can leave it if it's simply copied

```suggestion
/// NOTE(We need the root ArtifactOutput because of the Project type
/// but we are not using to compile anything zksync related)
#[derive(Debug)]
```